ledger/heavy/executor/initial_state_keeper_test.go

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2019 Insolar Technologies GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package executor
+
+import (
+	"testing"
+
+	"github.com/gojuno/minimock"
+	"github.com/insolar/insolar/insolar"
+	"github.com/insolar/insolar/insolar/gen"
+	"github.com/insolar/insolar/insolar/jet"
+	"github.com/insolar/insolar/insolar/record"
+	"github.com/insolar/insolar/instrumentation/inslogger"
+	"github.com/insolar/insolar/ledger/drop"
+	"github.com/insolar/insolar/ledger/object"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	pending insolar.PulseNumber = 780
+	topSync insolar.PulseNumber = 800
+	current insolar.PulseNumber = 850
+)
+
+func indexesFixture() []record.Index {
+	objIDs := gen.UniqueIDs(5)
+	return []record.Index{
+		{
+			ObjID: objIDs[0],
+			Lifeline: record.Lifeline{
+				EarliestOpenRequest: &pending,
+			},
+		},
+		{
+			ObjID: objIDs[1],
+			Lifeline: record.Lifeline{
+				EarliestOpenRequest: &pending,
+			},
+		},
+		{
+			ObjID: objIDs[2],
+			Lifeline: record.Lifeline{
+				EarliestOpenRequest: &pending,
+			},
+		},
+		{
+			ObjID: objIDs[3],
+			Lifeline: record.Lifeline{
+				EarliestOpenRequest: nil,
+			},
+		},
+		{
+			ObjID: objIDs[4],
+			Lifeline: record.Lifeline{
+				EarliestOpenRequest: nil,
+			},
+		},
+	}
+}
+
+func dropsFixture() []drop.Drop {
+	ids := gen.UniqueIDs(3)
+	return []drop.Drop{
+		{Split: false, Hash: ids[0].Hash()},
+		{Split: true, Hash: ids[1].Hash()},
+		{Split: false, Hash: ids[2].Hash()},
+	}
+}
+
+func TestInitialStateKeeper_Get_AfterRestart(t *testing.T) {
+	mc := minimock.NewController(t)
+	defer mc.Finish()
+	ctx := inslogger.TestContext(t)
+
+	topSyncPulse := insolar.Pulse{PulseNumber: topSync}
```

Comment:
why do you need `topSyncPulse` var? You only use `topSyncPulse.PulseNumber` value, which is just `topSync`